### DISCUSSION TITLE
Output slice data by fds2ascii

### DISCUSSION THREAD

**Discussion by wowwwt** [firemodels/fds#11867] [General]
The user guide shows us how to use fds2ascii to output the average value of some data over a period of time. But I want to output all  data in a certain time step over a period of time. I tried to change the source code to realize it, but failed. Could anyone give me some ideas? And as a novice, i want to ask some question: 1. What does [f2aversion] mean in the source code? Do I need to match the version of Pyrosim? 2. The executable file [fds2ascii] seems to have to be placed under the corresponding project file to run normally, so how can the script I modified in Visual Studio run normally?

**Comment by rmcdermo** [comment]
When fds2ascii is not sufficient, we recommend using the [slread.m](https://github.com/firemodels/fds/blob/master/Utilities/Matlab/scripts/slread.m) Matlab script.  There is also a python version [here](https://github.com/firemodels/fds/blob/master/Utilities/Python/scripts/slread.py).

**Reply by Firefighter1992** [reply]
Our team has developed an interactive program based on Python that calls fds2ascii to batch extract slice data. It appears to be capable of fulfilling your requirements. If you need it, please send an email to 718582618@qq.com. It's free.

**Reply by LordMicidial** [reply]
I'm trying to use slread.m, is there by any chance a tutorial or guide on how to plug into the proper directory and substitute the variables?  I have a file with over 20 Mesh and I have to use OCTAVE because MATLAB requires a license. My objective is being able to extract Temperatures from more than one Slices I've inserted in the simulation at specified Y= constant.

Do you think is going to be easier to try to install and use python? Any guidance on that file?

**Reply by rmcdermo** [reply]
Unfortunately, I have no experience with Octave.  We have instructions on installing python [here](https://github.com/firemodels/fds/wiki/Python-Setting-Up-Your-Environment).

We do not have much support for these scripts.  There are several packages being developed to work this problem.  See the "Data Readers" list in [Third-Party Tools](https://github.com/firemodels/fds/wiki/Third-Party-Tools).
